src/buffer/scroll_view.rs

```diff
@@ -22,20 +25,22 @@ pub enum Message {
     UserContext(user_context::Message),
     Link(message::Link),
     ScrollTo(keyed::Bounds),
+    RequestOlderChatHistory,
 }
 
 #[derive(Debug, Clone)]
 pub enum Event {
     UserContext(user_context::Event),
     OpenChannel(String),
     GoToMessage(Server, String, message::Hash),
+    RequestOlderChatHistory,
 }
 
 #[derive(Debug, Clone, Copy)]
 pub enum Kind<'a> {
     Server(&'a Server),
-    Channel(&'a Server, &'a str),
-    Query(&'a Server, &'a Nick),
+    Channel(&'a Server, &'a str, Option<ChatHistoryState>),
+    Query(&'a Server, &'a Nick, Option<ChatHistoryState>),
```

Comment:
Stuffing this into here seems to be causing us to require fetching this state when it's not even used. Checkout how we're querying for it when we construct `Kind` when calling `scroll_to_backlog`, this state isn't used anywhere there.

It seems we only need it during `view` so instead pass `Option<ChatHistoryState>` as a separate param there and it's just always `None` from server & query. This will clean up some code paths